src/Autoload.php

```diff
@@ -0,0 +1,202 @@
+			'requests_exception_http_415'                => '\WpOrg\Requests\Exception\Http\Status415',
+			'requests_exception_http_416'                => '\WpOrg\Requests\Exception\Http\Status416',
+			'requests_exception_http_417'                => '\WpOrg\Requests\Exception\Http\Status417',
+			'requests_exception_http_418'                => '\WpOrg\Requests\Exception\Http\Status418',
+			'requests_exception_http_428'                => '\WpOrg\Requests\Exception\Http\Status428',
+			'requests_exception_http_429'                => '\WpOrg\Requests\Exception\Http\Status429',
+			'requests_exception_http_431'                => '\WpOrg\Requests\Exception\Http\Status431',
+			'requests_exception_http_500'                => '\WpOrg\Requests\Exception\Http\Status500',
+			'requests_exception_http_501'                => '\WpOrg\Requests\Exception\Http\Status501',
+			'requests_exception_http_502'                => '\WpOrg\Requests\Exception\Http\Status502',
+			'requests_exception_http_503'                => '\WpOrg\Requests\Exception\Http\Status503',
+			'requests_exception_http_504'                => '\WpOrg\Requests\Exception\Http\Status504',
+			'requests_exception_http_505'                => '\WpOrg\Requests\Exception\Http\Status505',
+			'requests_exception_http_511'                => '\WpOrg\Requests\Exception\Http\Status511',
+			'requests_exception_http_unknown'            => '\WpOrg\Requests\Exception\Http\StatusUnknown',
+		);
+
+		/**
+		 * Whether or not a deprecation notice has been thrown for calling a deprecated class.
+		 *
+		 * @var bool
+		 */
+		private static $deprecation_thrown = false;
+
+		/**
+		 * Whether or not a deprecation notice should be thrown for calling a deprecated class.
+		 *
+		 * @var bool
+		 */
+		private static $throw_notice = true;
+
+		/**
+		 * Register the autoloader.
+		 *
+		 * Note: the autoloader is *prepended* in the autoload queue.
+		 * This is done to ensure that the Requests 2.0 autoloader takes precedence
+		 * over a potentially (dependency-registered) Requests 1.x autoloader.
+		 *
+		 * @internal This method contains a safeguard against the autoloader being
+		 * registered multiple times. This safeguard uses a global constant to
+		 * (hopefully/in most cases) still function correctly, even if the
+		 * class would be renamed.
+		 *
+		 * @return void
+		 */
+		public static function register() {
+			if (defined('REQUESTS_AUTOLOAD_REGISTERED') === false) {
+				spl_autoload_register(array(self::class, 'load'), true);
+				define('REQUESTS_AUTOLOAD_REGISTERED', true);
+			}
+		}
+
+		/**
+		 * Autoloader.
+		 *
+		 * @param string $class_name Name of the class name to load.
+		 *
+		 * @return bool Whether a class was loaded or not.
+		 */
+		public static function load($class_name) {
+			// Check that the class starts with "Requests" (PSR-0) or "WpOrg\Requests" (PSR-4).
+			$psr_4_prefix_pos = stripos($class_name, 'WpOrg\Requests\\');
```

Comment:
Inconsistency in escaping here:
```suggestion
			$psr_4_prefix_pos = stripos($class_name, 'WpOrg\\Requests\\');
```